rollup/core/classes/RollupRelationshipFieldFinder.cls

```diff
@@ -178,13 +178,13 @@ public without sharing class RollupRelationshipFieldFinder {
       ? this.metadata.GrandparentRelationshipFieldPath__c.split('\\.')
       : new List<String>();
 
-      if (localParts.isEmpty() && this.metadata.RollupToUltimateParent__c) {
-        String relationshipName = this.metadata.LookupFieldOnCalcItem__c.replace('__c', '__r').replace('Id', '');
-        localParts.add(relationshipName);
-        localParts.add(this.metadata.UltimateParentLookup__c);
-      }
+    if (localParts.isEmpty() && this.metadata.RollupToUltimateParent__c) {
+      String relationshipName = this.metadata.LookupFieldOnCalcItem__c.replace('__c', '__r').replace('Id', '');
```

Comment:
It looks like you've had this code in place for a while, but there's a slight chance that both of these `replace` statements won't work in some situations

- `replace('__c', '__r')` - this could have issues with namespaced fields, e.g. `SomeNamespace__customField__c` becomes `SomeNamespace__rustomObject__r`
- `replace('Id', '')` - this could have issues with custom fields that contain `Id` anywhere in the name, e.g. `Identity__c` becomes `entity__c`

Using something like `lastIndexOf(String)` to help narrow down where the replace should happen would help avoid those kinds of issues.

Reply:
mmm yes good call. I will fix this

Reply:
Fixed!